src/internal/coredb/projects.go

```diff
@@ -0,0 +1,80 @@
+package coredb
+
+import (
+	"context"
+	"fmt"
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"time"
+)
+
+// CreateProject creates an entry in the core.projects table.
+func CreateProject(ctx context.Context, tx *pachsql.Tx, project *pfs.ProjectInfo) error {
+	_, err := tx.ExecContext(ctx, "INSERT INTO core.projects (name, description) VALUES ($1, $2);", project.Project.Name, project.Description)
+	//todo: insert project.authInfo into auth table.
+	return errors.EnsureStack(err)
+}
+
+// DeleteProject deletes an entry in the core.projects table.
+func DeleteProject(ctx context.Context, tx *pachsql.Tx, projectName string) error {
+	_, err := tx.ExecContext(ctx, "DELETE FROM core.projects WHERE name = $1;", projectName)
+	//todo: delete corresponding project.authInfo auth table.
+	return errors.EnsureStack(err)
+}
+
+// GetProject retrieves an entry from the core.projects table by project name.
+func GetProject(ctx context.Context, tx *pachsql.Tx, projectName string) (*pfs.ProjectInfo, error) {
```

Comment:
I wonder if  read operations that are transactional should have a `Tx` suffix, like `GetProjectTx`.

Reply:
🤔 

I would say no because we only support one Get() function today which takes a transaction as a parameter. So that should be implicit in the behavior of the function. If we had multiple Get() functions, then maybe we could argue that their names should include `Tx`

Reply:
But I definitely see us potentially having a non-transactional Get() in the future.